core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -870,10 +866,7 @@ public void updateTask(TaskResult taskResult) {
                     task.getTaskDefName(), lastDuration, false, task.getStatus());
         }
 
-        // sync evaluate workflow only if the task is not within a forked branch
-        if (isLazyEvaluateWorkflow(workflowInstance.getWorkflowDefinition(), task)) {
-            expediteLazyWorkflowEvaluation(workflowId);
-        } else {
+        if (!isLazyEvaluateWorkflow(workflowInstance.getWorkflowDefinition(), task)) {
```

Comment:
@apanicker-nflx What's the intent of this change? It seems like an oversight cause if the condition is false (in other words, the workflow `isLazyEvalute`), nothing happens anymore.